GDJS/Runtime/variablescontainer.ts

```diff
@@ -192,15 +192,46 @@ namespace gdjs {
       const variable = this._variables.get(name);
       return !!variable && !variable.isUndefinedInContainer();
     }
+
+    /**
+     * Check if a variable exists in the container.
+     * @param variable The variable
+     * @return true if the variable exists.
+     */
+    hasVariable(variable: gdjs.Variable): boolean {
+      const foundVariable = this._variablesArray.find((v) => v === variable);
+      return !!foundVariable && !foundVariable.isUndefinedInContainer();
+    }
+
+    getVariableNameInContainer(variable: gdjs.Variable): string | undefined {
+      const index = this._variablesArray.indexOf(variable);
+      if (index === -1) {
+        return undefined;
+      }
+
+      const variableNames: string[] = [];
+      this._variables.keys(variableNames);
```

Comment:
This won't work, the keys order is not guaranteed by JS engines, so there is a high chance this breaks between browsers, JS engines or different runs of the game.
Instead, you should iterate probably directly of the hashmap and compare each value and if it's the variable you're searching, then we have the name as the key.